feat(SeaTable Node): Refactor, add new triggers and actions

christophdb opened the pull request:
# Complete rework of the SeaTable Node.

- Adding two new triggers and 12 new actions.
- Actions use now versioning
- cleaner code structure
- already published and tested as community node (n8n-nodes-seatable)

christophdb committed:
rework of seatable-n8n-node

christophdb committed:
bugfix wrong uploadurl for images/files

christophdb committed:
added missing loadOptionsDependsOn for update a row action

christophdb committed:
add new action "get many rows"

christophdb committed:
add case insensitive search to "search" action

christophdb committed:
fix "Make an API call"

christophdb committed:
improved "upload asset":

- input workspace_id is not required anymore
- two new options: `Replace existing file` and `Append to column`

Joffcom committed:
Merge branch 'master' of github.com:n8n-io/n8n into seatable_node_rework

Joffcom committed:
Lint fix on trigger node

Joffcom committed:
Fix breaking change that resulted in the old v1 node from appearing

christophdb committed:
supporting big data backend with links and row create, update and delete

christophdb committed:
take over change from Jeffcom

christophdb committed:
Fix error in upload action, if server domain ends with /

christophdb committed:
fixing asset upload

christophdb committed:
fixed linter errors and new action get row links

christophdb committed:
Merge branch 'master' of github.com:seatable/n8n into seatable_node_rework

christophdb committed:
Merge branch 'seatable_node_rework'

Joffcom commented:
Cheers @christophdb

Joffcom commented:
@christophdb Looks like there is still a lint issue that needs to be resolved, It may be worth removing some of the comments that are not needed as well.

christophdb committed:
fix lint issues and remove comments

christophdb committed:
fix typecheck errors

christophdb commented:
@Joffcom I am sorry, that my PR still had some issues. These "type check errors" were not raised by my VS Code. Now they should be fixed. Can you please try again? Thanks

Joffcom commented:
Hey @christophdb,

It looks like it is still reporting a bunch of lint issues, I have this pencilled in for review tomorrow morning.

netroy committed:
Merge remote-tracking branch 'origin/master' into seatable_node_rework2

netroy committed:
run lintfix

netroy committed:
revert documentationUrl

netroy committed:
revert icon svg file name

netroy commented:
I wanted to help fix this up, but unfortunately this PR is not editable by maintainers.
Can you please cherry-pick the last commits from [here](https://github.com/n8n-io/n8n/compare/master...seatable_node_rework2), so that we can get the review started on this 🙏🏽

netroy commented:
For any future PRs please leave this checkbox enabled 🙏🏽 
![image](https://github.com/n8n-io/n8n/assets/196144/ce33a9ff-358d-4825-add8-2a7623d50df6)

So that we can work with your PRs
![image](https://github.com/n8n-io/n8n/assets/196144/344157b9-b783-46ae-b0a7-117bb00bb624)

christophdb committed:
Merge branch 'master' into seatable_node_rework2

christophdb commented:
@netroy: I cherry-picked the commits. Please tell me if I can do something else.

Joffcom commented:
Overall I am happy with this, I will need to update our documentation when we merge this but I would like to take a moment to thank you @christophdb for taking on the work of making sure we are using the latest and greatest options that Seatable has to offer.

janober commented:
Thanks a lot, @christophdb and @Joffcom, for getting it done!

MehrCurry commented:
I am more than happy that this MR came back to life. A big thanks to all who made it possible.

michael-radency commented:
Hello @christophdb, thanks for your contribution
In the latest overhauls we moved from creating folder with three separate files for each operation to using single file `<operationName>.operation.ts` with two exports: description and execute.
Could you please consider updating your PR referring to v2 implementation of [Microsoft Outlook](https://github.com/n8n-io/n8n/tree/master/packages/nodes-base/nodes/Microsoft/Outlook/v2/actions) and [Airtable](https://github.com/n8n-io/n8n/tree/master/packages/nodes-base/nodes/Airtable/v2/actions/record) nodes

christophdb committed:
overhaul operation structure as requested by michael-radency

christophdb commented:
@michael-radency I updated the action folder with all operations as requested.

michael-radency commented:
> @michael-radency I updated the action folder with all operations as requested.

Thank you @christophdb , I'll try to review this PR today

christophdb commented:
@michael-radency Please don't merge now. My changes broke one of the loadOptionsDependsOn.

christophdb committed:
fix typo in listRows action

christophdb commented:
I found my copy-and-paste error and fixed it. Now you can merge it.

christophdb commented:
@michael-radency @Joffcom, is there anything else I can do at the moment?

michael-radency commented:
Hello @christophdb, code changes LGTM, but this overhaul requires more testing, there are some regression in v1 and I did encountered issues in v2

in v1 loaders are broken
![image](https://github.com/user-attachments/assets/e3b22321-ccc0-48f7-9674-97e6682b1ddd)
in v2 in some operations data not loaded, also ID shouldn't be appended to name
![image](https://github.com/user-attachments/assets/c9f04ba4-b640-4e5a-944f-433be5c396b2)
loaded in search though
![image](https://github.com/user-attachments/assets/2732db4a-d305-4db1-a0c3-62b5d0fe4840)

michael-radency commented on packages/nodes-base/nodes/SeaTable/v2/actions/router.ts:
```diff
@@ -0,0 +1,51 @@
+				responseData = await asset[seatable.operation].execute.call(this, i);
+			}
+
+			const executionData = this.helpers.constructExecutionMetaData(
+				this.helpers.returnJsonArray(responseData),
+				{ itemData: { item: i } },
+			);
```
`responseData` would be` INodeExecutionData[]` so can be simplified to
```
			const executionData = this.helpers.constructExecutionMetaData(responseData, {
				itemData: { item: i },
			});
```

christophdb commented:
I will have a look at these issues shortly.
@michael-radency: Just one question: how can I force n8n to create a node using the v1 version?

christophdb committed:
fix loader of v1

christophdb committed:
simplified router.ts

christophdb commented:
Hey @michael-radency,
thanks for your feedback. Here is what I did:

- broken v1 loader: 
fixed with commit 7d4f162

- in v2 in some operations data not loaded (found in asset - upload): 
the error message just says, that a column has to be selected. But the options are loaded correctly and available. As soon as a value is selected, everything is fine. What change would you expect here?
![image](https://github.com/user-attachments/assets/0887b79d-ce2e-4dab-9996-c16c70523b91)

- ID shouldn't be appended to name (found in asset - upload): 
I would like to keep this behaviour. From a technical perspective, I have to pass the ID:
![image](https://github.com/user-attachments/assets/7c9bf725-53f8-4be0-81cd-fe15594e0907)
But offering just the id as select options is not a good idea, because it is difficult for the user to determine the id from the web interface. On the other hand it is not a good idea to show only the name of the first column, because multiple rows can contain the name value. 
Therefore, I would like to keep the behaviour it like this:
![image](https://github.com/user-attachments/assets/92eb1fe6-def5-4bf5-8449-98c1af4ae9c3)

- loaded in search though (row - search): 
I don't have this issue. The loading is working fine. Error is just: value is required. What change would you expect here?
![image](https://github.com/user-attachments/assets/71c0c5c7-bf49-4271-abe0-5ea098bd4984)

- simplified router.ts:
I improved the code like you suggested.

Best regards
Christoph

christophdb commented:
Dear @netroy or @michael-radency,
Is there anything I can do?

christophdb commented:
Dear @netroy or @michael-radency,
sorry to bother you. Any updates on this or something that I can do?
Best regards
Christoph

Joffcom commented:
Hey @christophdb,

I am going to pick this one back up again during my next review session on Thursday morning.

christophdb committed:
remove faulty date input transformation with moment

Joffcom commented:
@christophdb it looks like there are some conflicts that need to be resolved again, I want to get this one merged in this quarter.

netroy committed:
Merge remote-tracking branch 'origin/master' into seatable_node_rework2

netroy committed:
update input/output types

netroy commented:
Since maintainers can't edit this PR, can you please cherry-pick the last 2 commits from [here](https://github.com/n8n-io/n8n/commits/seatable_node_rework2/) 🙏🏽

Joffcom commented:
Hey @christophdb,

Because we can't edit this PR and there are a lot of conflicts I have created a new PR with your changes here: https://github.com/n8n-io/n8n/pull/11431

This should allow us to work quickly to get this merged. I am going to close this one for now in favour of that one.
